oci-proxy: deploy sandbox infrastructure.

ameukam opened the pull request:
Add terraform code to handle deployment of a Cloud Run service for
`archeio`

- Add a global loadbalancer support IPv4 and IPv6
- ensure the Cloud Run run inside more than 1 GCP region
- ensure SSL certificates for `registry-sandbox.k8s.io`
- ensure traffic is redirect from HTTP to HTTPS

Still WIP because:
automated process of deployment is missing
discussion around naming

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

ameukam committed:
oci-proxy: Add infrastructure code for oci-proxy

Add terraform code to handle deployment of a Cloud Run service for
`archeio`

- Add a global loadbalancer support IPv4 and IPv6
- ensure the Cloud Run run inside more than 1 GCP region
- ensure SSL certificates for `registry-sandbox.k8s.io`
- ensure traffic is redirect from HTTP to HTTPS

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

ameukam commented:
cc @BenTheElder @endocrimes

ameukam commented:
Refs:
- https://github.com/kubernetes-sigs/oci-proxy/issues/13
- https://github.com/kubernetes/k8s.io/issues/3317

endocrimes commented on infra/gcp/terraform/k8s-infra-oci-proxy/network.tf:
```diff
@@ -0,0 +1,94 @@
+  #TODO(ameukam): current the TF resource google_compute_global_address don't have
+  #the value of IP in his attribute. But it's accessible with the data source:
+  #https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/compute_global_address
+  address      = "34.110.128.221"
```
this will have an unfortunate bootstrapping problem on a future clean bring-up, but that's probably ok for now. 

To avoid it, although it _kinda_ sucks from a guaranteeing convergence perspective in many cases, we could have a data source that depends on the IP address creation to look these up (Or we end up refactoring this to work in multiple steps).

ameukam committed:
bump archeio to the latest tag

Changes:
- https://github.com/kubernetes-sigs/oci-proxy/pull/14
- https://github.com/kubernetes-sigs/oci-proxy/pull/15

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

ameukam commented:
/retitle oci-proxy: deploy sandbox infrastructure.

I think we can merge this for the moment.

ameukam commented:
/hold cancel
